ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/kre/base/kase/KreCaseSubscriptionTest.java

```diff
@@ -1,9 +1,307 @@
-package com.kayako.android.k5.push.kre.base.kase;
+package com.kayako.sdk.android.k5.kre.base.kase;
 
-import static org.junit.Assert.*;
+import com.kayako.sdk.android.k5.kre.base.KreConnection;
+import com.kayako.sdk.android.k5.kre.base.KreSubscription;
+import com.kayako.sdk.android.k5.kre.base.credentials.KreCredentials;
+import com.kayako.sdk.android.k5.kre.helpers.MinimalClientTypingListener;
+import com.kayako.sdk.android.k5.kre.helpers.RawCaseChangeListener;
+import com.kayako.sdk.android.k5.kre.helpers.RawCasePostChangeListener;
+import com.kayako.sdk.android.k5.kre.helpers.RawClientActivityListener;
+import com.kayako.sdk.android.k5.kre.helpers.RawClientTypingListener;
+import com.kayako.sdk.android.k5.kre.helpers.RawUserOnCasePresenceListener;
+import com.kayako.sdk.android.k5.kre.helpers.presence.KrePresenceHelper;
+import static org.hamcrest.CoreMatchers.is;
+import org.hamcrest.Matcher;
+import org.hamcrest.core.IsNull;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ErrorCollector;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.support.membermodification.MemberMatcher.method;
+import static org.powermock.api.support.membermodification.MemberModifier.suppress;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+import java.util.List;
 
+@PrepareForTest(KreSubscription.class)
+@RunWith(PowerMockRunner.class)
 public class KreCaseSubscriptionTest {
 
-    // TODO: Ensure that triggerIsViewing is called onSubscription - why? - because with the new kRE rules, the app user is not considered actively viewing a case unless he triggers a isViewing isForeground push event
+    private static final String NAME = "TEST_NAME";
+    private static final long USER_ID = 1L;
```

Comment:
✅ Good to see constant used instead of hard-coded values.

This approach increases maintainability (single point of modification) and debugging, avoid duplication and decreases verbosity.

**No change suggested, looks good!**	  

<a id="a2c85680-4980-11e8-86d4-9358f2b1b645" href="none" codereview-uuid-reserved-tag />